internal/alpini/netty4/alpini-netty4-base/src/main/java/com/linkedin/alpini/netty4/ssl/SslInitializer.java

```diff
@@ -577,9 +591,19 @@ public void operationComplete(Future<String> future) {
         }
 
         private void resolved(Future<? super Void> future) {
+          // Step 3.5: This function is called after DNS resolution completes (if _resolveClient enabled); next step is
+          // to replace itself with SSL handler. If _resolveClient not enabled, SslDetect would be replaced
+          // with SSL handler at the decode function directly, before it has a chance to reach this function.
           // This is executed on the channel executor thread
           assert _channelHandlerContext.channel().eventLoop().inEventLoop();
 
+          Long dnsStartTime = (Long) _channelHandlerContext.channel().attr(DNS_START_TS).get();
+          long dnsLatencyMs = TimeUnit.NANOSECONDS.toMillis(Time.nanoTime() - dnsStartTime);
+          LOG.log(
```

Comment:
Can we add a debug in func: `initChannel`? That is supposed to be the first endpoint before starting ssl handshake.

Reply:
Let me see if I can get some information about the channel into the log.